web/models/users.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Notox.User do
+  use Notox.Web, :model
+
+  @required_fields [
+    :username,
+  ]
+
+  @optional_fields @required_fields
+
+  schema "users" do
+    field :username, :string
+    field :password, :string, virtual: true
+    field :password_hash, :string
+
+    timestamps
+  end
+
+  def changeset(model, params \\ %{}) do
+    model
+    |> cast(params, @optional_fields)
+    |> validate_required(@required_fields)
+  end
+
+  defp put_pass_hash(changeset) do
+    case changeset do
+      %Ecto.Changeset{valid?: true, changes: %{password: password}} ->
+        put_change(changeset, :password_hash, Comeonin.Bcrypt.hashpwsalt(password))
+      _ ->
+        changeset
+    end
+  end
+
+  def registration_changeset(model, params \\ %{}) do
+    model
+    |> changeset(params)
+    |> unique_constraint(:username)
```

Comment:
This should be in `changeset`, not `registration_changeset`. As it stands the an internal error would be caused by editing a username so that it's not unique.